src/main.jl

```diff
@@ -176,6 +176,23 @@ type DefaultUnknown <: UnknownCategory
 end
 
 @doc """
+Categories of constraints on Unknowns; used to create positive, negative, etc., constraints.
+""" ->
+abstract UnknownConstraint
+
+
+type Normal <: UnknownConstraint
```

Comment:
These all need documentation strings because they are exported (a oneliner is fine).